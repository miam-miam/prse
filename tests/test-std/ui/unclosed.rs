use prse::parse;

fn main() {
    let l = "test: 5";

    let mut x = 0;

    parse!(l, "test: {x");
    let y: i32 = parse!(l, "test: {");

    parse!(l, "test: x}");
    let y: i32 = parse!(l, "test: }");

    assert_eq!(5_i32, x);
}
