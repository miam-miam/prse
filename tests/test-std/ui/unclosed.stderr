error: Expected to find } bracket. Consider adding a } bracket to close the open { bracket.
 --> ui/unclosed.rs:8:15
  |
8 |     parse!(l, "test: {x");
  |               ^^^^^^^^^^

error: Expected to find } bracket. Consider adding a } bracket to close the open { bracket.
 --> ui/unclosed.rs:9:28
  |
9 |     let y: i32 = parse!(l, "test: {");
  |                            ^^^^^^^^^

error: Found unexpected } bracket. Consider escaping it by changing it to }}.
  --> ui/unclosed.rs:11:15
   |
11 |     parse!(l, "test: x}");
   |               ^^^^^^^^^^

error: Found unexpected } bracket. Consider escaping it by changing it to }}.
  --> ui/unclosed.rs:12:28
   |
12 |     let y: i32 = parse!(l, "test: }");
   |                            ^^^^^^^^^
